pkg/genericapiserver/genericapiserver.go

```diff
@@ -583,10 +583,24 @@ func verifyEtcdServersList(options *options.ServerRunOptions) {
 	}
 }
 
+func verifySecurePort(options *options.ServerRunOptions) {
+	if options.SecurePort < 1 || options.SecurePort > 65535 {
+		glog.Fatalf("--secure-port %v must be between 1 and 65535, inclusive", options.SecurePort)
+	}
+}
+
+func verifyInsecurePort(options *options.ServerRunOptions) {
+	if options.InsecurePort < 1 || options.InsecurePort > 65535 {
```

Comment:
I'm not sure if this works now, but I'd like for 0 to turn off the insecure port. Maybe add a TODO if that's not how it works now.